server.js

```diff
@@ -6030,6 +6030,91 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Uptime Robot status integration.
+camp.route(/^\/uptimerobot\/status\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var monitorApiKey = match[1];  // eg, u956-afus321g565fghr519
+  var format = match[2];
+  var badgeData = getBadgeData('status', data);
+  var options = {
+    method: 'POST',
+    json: true,
+    body: {
+      "api_key": monitorApiKey,
+      "format": "json"
+    },
+    uri: 'https://api.uptimerobot.com/v2/getMonitors'
+  };
+  request(options, function(err, res, json) {
+    if (err !== null || res.statusCode >= 500 || typeof json !== 'object') {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
```

Comment:
Thanks for adding that. Looks great. Though, I think it would be better to check that `json.error.message` is a string. Otherwise if `error` is an empty object you could end up with `undefined` on the badge.

```
if (json.error && typeof json.error.message === 'string') {
  badgeData.text[1] = json.error.message;
} else {
  badgeData.text[1] = 'vendor error';
}
```